### PR TITLE
Clean up registration policies and separate design and implementation

### PR DESCRIPTION
**Pull request by ad-l**

### CODE REVIEW

**Review by fournet — APPROVED** [review]

File: draft-birkholz-scitt-architecture.md [hunk 1]
```diff
@@ -454,6 +442,24 @@ Once the Statement is serialized with the correct content type, the Issuer shoul
...
 
+## Standard registration policies
+
+> **Editor's note**
+>
+> The technical design for signalling and verifying registration policies is a work in progress.
+> We expect that once the formats and semantics of the registration policy headers are finalized, standardized policies may be moved to a separate draft.
+> For now, we inline some significant policies to illustrate the most common use cases.
+
+All TS implementations MUST support the following registration policies, and MUST check that all the policies defined in the `reg_info` map are satisfied before a claim can be registered. If the `reg_info` map of a claim contains an unknown policy entry, the claim MUST be rejected. This is to ensure that all verifiers get the same guarantee out of the registration policies regardless of where it is registered.
```
Comment (letmaik) [inline draft-birkholz-scitt-architecture.md]: This is too strong. We can't expect all implementations to support all listed policies.

File: draft-birkholz-scitt-architecture.md [hunk 2]
```diff
@@ -298,32 +298,20 @@ For example, consider a TS implemented using a set of replicas, each running wit
...
-
-While Registration policies are a burden for Issuers (some may require them to maintain state to remember what they have signed before) they support stronger transparency guarantees, and they greatly help Verifiers and auditors in making sense of the information on the Registry. (This is particularly relevant for parties that verify Receipts on their own, without accessing the Registry). For instance, if a TS doesn't apply any policy, Claims may be registered in a different order than they have been issued, and old Claims may be replayed, which makes it difficult to understand the logical history of an Artifact, or to prevent rollback attacks.
+A TS that accepts to register any valid claim offers limited value to verifiers: the baseline transparency guarantee ensures completeness of audit, which can help detect equivocation.
```
Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
A TS that accepts to register any valid claim offered by an issuer would end up providing only limited value to verifiers. In consequence, a baseline transparency guarantee policing the registration of claims is required to ensure completeness of audit, which can help detect equivocation.
```

File: draft-birkholz-scitt-architecture.md [hunk 3]
```diff
@@ -298,32 +298,20 @@ For example, consider a TS implemented using a set of replicas, each running wit
...
+Most advanced SCITT scenarios rely on the TS performing additional domain-specific checks before a claim is accepted: they may only allow trusted authenticated users to register claims, they may try to check the new claim is consistent with previous claims from the same issuers or that they are registered in the correct order and cannot be re-played; some TS may even interpret and validate the payload of claims.
```
Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
Most advanced SCITT scenarios rely on the TS performing additional domain-specific checks before a claim is accepted: TS may only allow trusted authenticated users to register claims, TS may try to check that a new claim is consistent with previous claims from the same issuers or that claims are registered in the correct order and cannot be re-played; some TS may even interpret and validate the payload of claims.
```

File: draft-birkholz-scitt-architecture.md [hunk 4]
```diff
@@ -298,32 +298,20 @@ For example, consider a TS implemented using a set of replicas, each running wit
...
 
-There are two kinds of Registration policies: (1) named policies have standardized semantics that are uniform across all implementations of SCITT Transparency Services, while (2) custom policies are opaque and may contain pointers to (or even inlined) policy descriptions (declarative or programmable).
+In general, registration policies are applied at the discretion of the TS, and verifiers use the receipt as a witness that the registration policy of the TS was satisfied at the time the claim was registered. Transparency service implementations SHOULD make their full registration policy public and auditable, e.g. by recording the stateful policy inputs at evaluation time in the registry to ensure the policy can be later independently validated. However, from an interoperability point of view the policy that was applied by the TS is opaque to the verifier, who is forced to trust the associated registration policy. If the policy of the TS evolves over time, or is different across issuers, the guarantee derived from receipt validation may not be uniform across all claims.
```
Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
In general, registration policies are applied at the discretion of the TS, and verifiers use receipts as witnesses that confirm that the registration policy of the TS was satisfied at the time claim registration. TS implementations SHOULD make their full registration policy public and auditable, e.g. by recording stateful policy inputs at evaluation time in the registry to ensure that policy can be independently validated later. 
From an interoperability point of view, the policy that was applied by the TS is opaque to the verifier, who is forced to trust the associated registration policy. If the policy of the TS evolves over time, or is different across issuers, the guarantee derived from receipt validation may not be uniform across all claims over time.
```

File: draft-birkholz-scitt-architecture.md (hunk 1 above)
Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
TS implementations MUST indicate their support for registration policies and MUST check that all the policies indicated as defined in the `reg_info` map are supported and are satisfied before a claim can be registered. Any unsupported claims MUST be indicated separately and corresponding unknown policy entries in the map of a claim MUST be rejected. This is to ensure that all verifiers get the same guarantee out of the registration policies regardless of where it is registered.
```

Maik, et al. Does this improve the wording?

**Review by henkbirkholz — CHANGES_REQUESTED** [review]

File: draft-birkholz-scitt-architecture.md (hunk 2 above)
Comment (ad-l) [reply draft-birkholz-scitt-architecture.md]: Agree with the first change but not with the second - audit completeness is the only property that *does not* depend on the registration policy, just the fact that the claim is in the append-only registry.

```suggestion
A TS that accepts to register any valid claim offered by an issuer only provides the baseline audit completeness of transparency to verifiers., which may help detect equivocation but is not enough to filter and police bad claims.
```

File: draft-birkholz-scitt-architecture.md (hunk 3 above)
Comment (ad-l) [reply draft-birkholz-scitt-architecture.md]: +1

File: draft-birkholz-scitt-architecture.md (hunk 4 above)
Comment (ad-l) [reply draft-birkholz-scitt-architecture.md]: +1

File: draft-birkholz-scitt-architecture.md [hunk 5]
```diff
@@ -454,6 +443,24 @@ Once the Statement is serialized with the correct content type, the Issuer shoul
...
 
+## Standard registration policies
+
+> **Editor's note**
+>
+> The technical design for signalling and verifying registration policies is a work in progress.
+> We expect that once the formats and semantics of the registration policy headers are finalized, standardized policies may be moved to a separate draft.
+> For now, we inline some significant policies to illustrate the most common use cases.
+
+All TS implementations MAY only support a subset of the standard policies defined below, but it MUST enforce that all supported policies that appear in the `reg_info` map are satisfied before a claim can be registered and it MUST reject the claim if it contains any unknown or unsupported policy. This ensures that all verifiers get the same guarantee out of the registration policies regardless of where it is registered.
```
Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md:454]:
```suggestion
All TS implementations MAY only support a subset of the standard policies defined below, but it MUST enforce that all supported policies that appear in the `reg_info` map are satisfied before a claim can be registered and it MUST reject the claim if it contains any unknown or unsupported policy. This ensures that all verifiers get the same guarantee out of the registration policies regardless of where it is registered.
```

Comment (henkbirkholz) [inline draft-birkholz-scitt-architecture.md]:
```suggestion
TS implementations MUST indicate their support for registration policies and MUST check that all the policies indicated as defined in the `reg_info` map are supported and are satisfied before a claim can be registered. Any unsupported claims MUST be indicated separately and corresponding unknown policy entries in the map of a claim MUST be rejected. This is to ensure that all verifiers get the same guarantee out of the registration policies regardless of where it is registered.
```

**Pull request merged**